pkg/rpc/rpc_handler.go

```diff
@@ -95,6 +125,44 @@ func (s *server) EnforceNpToPod(ctx context.Context, in *rpc.EnforceNpRequest) (
 	return &resp, nil
 }
 
+// DeletePodNp processes CNI Delete Pod NP network request
+func (s *server) DeletePodNp(ctx context.Context, in *rpc.DeleteNpRequest) (*rpc.DeleteNpReply, error) {
+	if s.policyReconciler.GeteBPFClient() == nil {
+		s.log.Info("Network policy is disabled, returning success")
+		success := rpc.DeleteNpReply{
+			Success: true,
+		}
+		return &success, nil
+	}
+
+	s.log.Info("Received Delete Network Policy Request for Pod", "Name", in.K8S_POD_NAME, "Namespace", in.K8S_POD_NAMESPACE)
+	var err error
+	podIdentifier := utils.GetPodIdentifier(in.K8S_POD_NAME, in.K8S_POD_NAMESPACE, s.log)
+
+	value, _ := s.policyReconciler.GeteBPFClient().GetDeletePodLockMap().LoadOrStore(podIdentifier, &sync.Mutex{})
+	deletePodLock := value.(*sync.Mutex)
+	deletePodLock.Lock()
+	s.log.Info("Got the deletePodLock for", "Pod: ", in.K8S_POD_NAME, " Namespace: ", in.K8S_POD_NAMESPACE, " PodIdentifier: ", podIdentifier)
+
+	isProgFdShared, err := s.policyReconciler.IsProgFdShared(in.K8S_POD_NAME, in.K8S_POD_NAMESPACE)
```

Comment:
ok, this answers one of my previous comments. So we capture the delete pod event and cleanup bpf progs and maps..